src/main/java/teammates/storage/search/CommentSearchQuery.java

```diff
@@ -6,42 +6,32 @@
 import teammates.common.util.Const;
 import teammates.logic.core.InstructorsLogic;
 
-import com.google.appengine.api.search.Cursor;
 import com.google.appengine.api.search.Document;
-import com.google.appengine.api.search.QueryOptions;
 
 /**
  * The SearchQuery object that defines how we query {@link Document} for student comments
  */
-public class CommentSearchQuery extends SearchQuery {
+public class CommentSearchQuery extends BaseCommentSearchQuery {
+    
     public CommentSearchQuery(String googleId, String queryString, String cursorString) {
-        Cursor cursor = cursorString.isEmpty()
-                ? Cursor.newBuilder().build()
-                : Cursor.newBuilder().build(cursorString);
-        QueryOptions options = QueryOptions.newBuilder()
-                .setLimit(20)
-                .setCursor(cursor)
-                .build();
-        setOptions(options);
-        prepareVisibilityQueryString(googleId);
-        setTextFilter(Const.SearchDocumentField.SEARCHABLE_TEXT, queryString);
+        super(googleId, queryString, cursorString);
     }
-    
-    private void prepareVisibilityQueryString(String googleId){
+
+    protected StringBuilder prepareVisibilityQueryString(String googleId) {
+        StringBuilder courseIdLimit = super.prepareVisibilityQueryString(googleId);
         List<InstructorAttributes> instructorRoles = InstructorsLogic.inst().getInstructorsForGoogleId(googleId);
```

Comment:
InstructorsLogic.inst().getInstructorsForGoogleId(googleId) is called twice.
be careful of this kind of issue when dealing with inheritance things